// @opuscapita/oc-cm-common-layouts/lib/cjs/primitives/index.js
'use strict';

exports.__esModule = true;
exports.Spinner = exports.Block = exports.Subtitle = exports.Title = exports.Header = exports.Input = exports.Button = exports.BorderlessButton = undefined;

var _primitiveBorderlessButton = require('./primitive-borderless-button.component');

var _primitiveBorderlessButton2 = _interopRequireDefault(_primitiveBorderlessButton);

var _primitiveButton = require('./form-controls/primitive-button.component');

var _primitiveButton2 = _interopRequireDefault(_primitiveButton);

var _primitiveInput = require('./form-controls/primitive-input.component');

var _primitiveInput2 = _interopRequireDefault(_primitiveInput);

var _primitiveHeader = require('./primitive-header.component');

var _primitiveHeader2 = _interopRequireDefault(_primitiveHeader);

var _primitiveTitle = require('./primitive-title.component');

var _primitiveTitle2 = _interopRequireDefault(_primitiveTitle);

var _primitiveSubtitle = require('./primitive-subtitle.component');

var _primitiveSubtitle2 = _interopRequireDefault(_primitiveSubtitle);

var _primitiveBlock = require('./primitive-block.component');

var _primitiveBlock2 = _interopRequireDefault(_primitiveBlock);

var _primitiveSpinner = require('./primitive-spinner.component');

var _primitiveSpinner2 = _interopRequireDefault(_primitiveSpinner);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.BorderlessButton = _primitiveBorderlessButton2.default;
exports.Button = _primitiveButton2.default;
exports.Input = _primitiveInput2.default;
exports.Header = _primitiveHeader2.default;
exports.Title = _primitiveTitle2.default;
exports.Subtitle = _primitiveSubtitle2.default;
exports.Block = _primitiveBlock2.default;
exports.Spinner = _primitiveSpinner2.default;
