// @opuscapita/react-floating-select/lib/cjs/floating-select-windowed/floating-select-windowed-menu-list.component.js
'use strict';

exports.__esModule = true;

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactWindow = require('react-window');

var _util = require('./util');

var _styles = require('../floating-select/styles');

var _styles2 = _interopRequireDefault(_styles);

require('./floating-select-windowed.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// Other imports


var MenuList = function (_React$PureComponent) {
  _inherits(MenuList, _React$PureComponent);

  function MenuList(props) {
    _classCallCheck(this, MenuList);

    var _this = _possibleConstructorReturn(this, _React$PureComponent.call(this, props));

    _this.renderOption = function (_ref) {
      var data = _ref.data,
          index = _ref.index,
          baseStyle = _ref.baseStyle;

      var inputHeight = _this.props.selectProps.inputHeight || '30px';
      var customStyles = (0, _styles2.default)(inputHeight);
      var combinedStyles = customStyles.option(baseStyle, data[index].props);

      return _react2.default.createElement(
        'div',
        {
          className: 'floating-select-windowed-menu-option',
          style: combinedStyles
        },
        data[index]
      );
    };

    _this.setListRef = _this.setListRef.bind(_this);
    _this.getItemSize = _this.getItemSize.bind(_this);

    _this.state = {
      currentIndex: 0,
      children: null,
      heights: [],
      itemCount: 0,
      menuHeight: 0
    };
    return _this;
  }

  MenuList.getDerivedStateFromProps = function getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.children !== prevState.children) {
      var getStyles = nextProps.getStyles;

      var children = _react2.default.Children.toArray(nextProps.children);

      var head = children[0] || {};
      var _head$props = head.props;
      _head$props = _head$props === undefined ? {} : _head$props;
      var _head$props$data = _head$props.data;
      _head$props$data = _head$props$data === undefined ? {} : _head$props$data;
      var _head$props$data$opti = _head$props$data.options,
          options = _head$props$data$opti === undefined ? [] : _head$props$data$opti;

      var groupedChildrenLength = options.length;
      var isGrouped = groupedChildrenLength > 0;
      var flattenedChildren = isGrouped && (0, _util.flattenGroupedChildren)(children);

      children = isGrouped ? flattenedChildren : children;

      var groupHeadingStyles = getStyles('groupHeading', nextProps);
      var loadingMsgStyles = getStyles('loadingMessage', nextProps);
      var noOptionsMsgStyles = getStyles('noOptionsMessage', nextProps);
      var optionStyles = getStyles('option', nextProps);
      var getHeight = (0, _util.createGetHeight)({
        groupHeadingStyles: groupHeadingStyles,
        noOptionsMsgStyles: noOptionsMsgStyles,
        optionStyles: optionStyles,
        loadingMsgStyles: loadingMsgStyles
      });

      var heights = children.map(getHeight);

      var currentIndex = (0, _util.getCurrentIndex)(children);

      var itemCount = children.length;

      // calc menu height
      var sum = function sum(a, b) {
        return a + b;
      };
      var totalHeight = heights.reduce(sum, 0);

      var _getStyles = getStyles('menuList', nextProps),
          maxHeight = _getStyles.maxHeight;

      var menuHeight = Math.min(maxHeight, totalHeight);
      var estimatedItemSize = Math.floor(totalHeight / itemCount);

      return {
        children: children,
        currentIndex: currentIndex,
        estimatedItemSize: estimatedItemSize,
        heights: heights,
        itemCount: itemCount,
        menuHeight: menuHeight
      };
    }

    return null;
  };

  MenuList.prototype.componentDidUpdate = function componentDidUpdate() {
    var currentIndex = this.state.currentIndex;


    if (this.state.children.length === 1) {
      this.list.resetAfterIndex(0);
    }

    /**
     * enables scrolling on key down arrow
     *
     * note: prevents scrolling on index 0 and 1 to avoid
     * returning to top of menu when it remains open after selecting
     */
    if (currentIndex > 1) {
      this.list.scrollToItem(currentIndex);
    }
  };

  MenuList.prototype.getItemSize = function getItemSize(index) {
    return this.state.heights[index];
  };

  MenuList.prototype.setListRef = function setListRef(ref) {
    this.list = ref;
  };

  MenuList.prototype.render = function render() {
    var _props = this.props,
        getStyles = _props.getStyles,
        innerRef = _props.innerRef;
    var _state = this.state,
        stateChildren = _state.children,
        estimatedItemSize = _state.estimatedItemSize,
        menuHeight = _state.menuHeight,
        itemCount = _state.itemCount;

    var _getStyles2 = getStyles('menuList', this.props),
        maxHeight = _getStyles2.maxHeight,
        menuListStyle = _objectWithoutProperties(_getStyles2, ['maxHeight']);

    return _react2.default.createElement(
      'div',
      { ref: innerRef, style: menuListStyle, className: 'floating-select-windowed-menu-list' },
      _react2.default.createElement(
        _reactWindow.VariableSizeList,
        {
          ref: this.setListRef,
          estimatedItemSize: estimatedItemSize,
          height: menuHeight,
          itemCount: itemCount,
          itemData: stateChildren,
          itemSize: this.getItemSize
        },
        this.renderOption
      )
    );
  };

  return MenuList;
}(_react2.default.PureComponent);

exports.default = MenuList;
